src/main/java/seedu/sellsavvy/model/order/OrderList.java

```diff
@@ -53,11 +54,28 @@ public void add(Order toAdd) {
      * Removes the first instance of an equivalent order from the list.
      * The order must exist in the list.
      */
-    public void remove(Order toRemove) {
-        requireNonNull(toRemove);
-        if (!internalList.remove(toRemove)) {
+    public void remove(Index index) {
+        requireNonNull(index);
+        if (index.getZeroBased() >= this.size()) {
             throw new OrderNotFoundException();
         }
+        internalList.remove(index.getZeroBased());
+    }
+
+    /**
+     * Retrieves an order from the list at the specified index.
+     * The index must be less than the list size.
+     */
+    public Order get(int index) {
+        return internalList.get(index);
+        // This will automatically throw IndexOutOfBoundsException if the index is invalid.
+    }
```

Comment:
Given that remove(Index) already handles order removal, is this method still necessary?

Reply:
The `DeleteOrderCommand` still needs to get the order to include it under the `MESSAGE_DELETE_ORDER_SUCCESS` message.
```
        Order orderToDelete = unmodifiableOrderList.get(this.index.getZeroBased());
        orderList.remove(this.index);
        return new CommandResult(String.format(MESSAGE_DELETE_ORDER_SUCCESS, Messages.format(orderToDelete)));
```

Reply:
Ok I will kiv for future commits